File: leetcode/121.best-time-to-buy-and-sell-stock.py
#
# @lc app=leetcode id=121 lang=python3
#
# [121] Best Time to Buy and Sell Stock
#

# @lc code=start
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        buy = 10 ** 4 + 1
        for price in prices:
            if buy > price:
                buy = price
            ans = max(ans, price - buy)
        return ans


# @lc code=end

File: leetcode/121.best-time-to-buy-and-sell-stock.java
/*
 * @lc app=leetcode id=121 lang=java
 *
 * [121] Best Time to Buy and Sell Stock
 */

// @lc code=start
class Solution {
    public int maxProfit(int[] prices) {
        int min = (int) Math.pow(10, 4);
        int max = 0;
        for (int p : prices) {
            if (p < min) {
                min = p;
                continue;
            }
            max = Math.max(max, p - min);

        }

        return max;

    }
}
// @lc code=end
